File: archive/mobile-app-ios-legacy/Plagit/Candidate/CVReviewView.swift
//
//  CVReviewView.swift
//  Plagit
//
//  Review screen for CV-extracted profile data.
//  Shows extracted fields, allows editing, then saves to profile.
//

import SwiftUI

struct CVReviewView: View {
    @Environment(\.dismiss) private var dismiss
    let extractedData: CandidateAPIService.CVExtractedData
    var onSave: (CVReviewView.ReviewedFields) async -> Void

    struct ReviewedFields {
        var name: String
        var phone: String
        var location: String
        var role: String
        var experience: String
        var languages: String
        var bio: String
        var jobType: String
        var email: String
        var skills: String
        var certifications: String
    }

    @State private var firstName = ""
    @State private var lastName = ""
    @State private var email = ""
    @State private var phone = ""
    @State private var location = ""
    @State private var role = ""
    @State private var experience = ""
    @State private var languages = ""
    @State private var skills = ""
    @State private var certifications = ""
    @State private var bio = ""
    @State private var jobType = "Full-time"
    @State private var isSaving = false

    private let jobTypes = ["Full-time", "Part-time", "Contract", "Freelance", "Temporary", "Flexible"]

    var body: some View {
        NavigationStack {
            ZStack {
                Color.plagitBackground.ignoresSafeArea()
                ScrollView(.vertical, showsIndicators: false) {
                    VStack(alignment: .leading, spacing: PlagitSpacing.xl) {

                        // Header
                        VStack(spacing: PlagitSpacing.sm) {
                            ZStack {
                                Circle().fill(Color.plagitOnline.opacity(0.1)).frame(width: 56, height: 56)
                                Image(systemName: "checkmark.circle.fill")
                                    .font(.system(size: 28, weight: .medium))
                                    .foregroundColor(.plagitOnline)
                            }
                            Text("We extracted the details below from your CV. Please review, edit if needed, and confirm before saving.")
                                .font(PlagitFont.caption())
                                .foregroundColor(.plagitSecondary)
                                .multilineTextAlignment(.center)
                            Text("Nothing will be saved until you confirm.")
                                .font(PlagitFont.micro())
                                .foregroundColor(.plagitTertiary)
                        }
                        .frame(maxWidth: .infinity)
                        .padding(.top, PlagitSpacing.lg)

                        // Fields
                        fieldCard("Personal Details") {
                            editField("First Name", text: $firstName, icon: "person")
                            editField("Last Name", text: $lastName, icon: "person")
                            editField("Phone Number", text: $phone, icon: "phone", keyboard: .phonePad)
                            editField("Location", text: $location, icon: "mappin")
                        }

                        fieldCard("Professional") {
                            editField("Job Role", text: $role, icon: "briefcase")
                            editField("Years of Experience", text: $experience, icon: "clock")

                            // Job type picker
                            VStack(alignment: .leading, spacing: PlagitSpacing.xs) {
                                HStack(spacing: PlagitSpacing.xs) {
                                    Image(systemName: "person.badge.clock").font(.system(size: 12, weight: .medium)).foregroundColor(.plagitTeal)
                                    Text("Job Type").font(PlagitFont.captionMedium()).foregroundColor(.plagitSecondary)
                                }
                                ScrollView(.horizontal, showsIndicators: false) {
                                    HStack(spacing: PlagitSpacing.sm) {
                                        ForEach(jobTypes, id: \.self) { type in
                                            let active = jobType == type
                                            Button { jobType = type } label: {
                                                Text(type)
                                                    .font(PlagitFont.captionMedium())
                                                    .foregroundColor(active ? .white : .plagitSecondary)
                                                    .fixedSize()
                                                    .padding(.horizontal, PlagitSpacing.md)
                                                    .padding(.vertical, PlagitSpacing.sm)
                                                    .background(Capsule().fill(active ? Color.plagitTeal : Color.plagitSurface))
                                            }
                                        }
                                    }
                                }
                            }
                        }

                        fieldCard("Languages & Skills") {
                            editField("Languages", text: $languages, icon: "globe")
                            editField("Skills", text: $skills, icon: "star")
                            editField("Certifications", text: $certifications, icon: "rosette")
                        }

                        if !bio.isEmpty {
                            fieldCard("About") {
                                VStack(alignment: .leading, spacing: PlagitSpacing.xs) {
                                    HStack(spacing: PlagitSpacing.xs) {
                                        Image(systemName: "text.alignleft").font(.system(size: 12, weight: .medium)).foregroundColor(.plagitTeal)
                                        Text("Bio").font(PlagitFont.captionMedium()).foregroundColor(.plagitSecondary)
                                    }
                                    TextEditor(text: $bio)
                                        .font(PlagitFont.body())
                                        .foregroundColor(.plagitCharcoal)
                                        .frame(minHeight: 60)
                                        .scrollContentBackground(.hidden)
                                }
                            }
                        }

                        // Edit note
                        Text("You can edit any extracted field before saving.")
                            .font(PlagitFont.micro())
                            .foregroundColor(.plagitTertiary)
                            .frame(maxWidth: .infinity)

                        // Save button
                        Button {
                            Task {
                                isSaving = true
                                let fullName = [firstName, lastName].map { $0.trimmingCharacters(in: .whitespaces) }.filter { !$0.isEmpty }.joined(separator: " ")
                                await onSave(ReviewedFields(
                                    name: fullName,
                                    phone: phone.trimmingCharacters(in: .whitespaces),
                                    location: location.trimmingCharacters(in: .whitespaces),
                                    role: role.trimmingCharacters(in: .whitespaces),
                                    experience: experience.trimmingCharacters(in: .whitespaces),
                                    languages: languages.trimmingCharacters(in: .whitespaces),
                                    bio: bio.trimmingCharacters(in: .whitespaces),
                                    jobType: jobType,
                                    email: email.trimmingCharacters(in: .whitespaces),
                                    skills: skills.trimmingCharacters(in: .whitespaces),
                                    certifications: certifications.trimmingCharacters(in: .whitespaces)
                                ))
                                isSaving = false
                                dismiss()
                            }
                        } label: {
                            HStack(spacing: PlagitSpacing.sm) {
                                if isSaving {
                                    ProgressView().tint(.white)
                                } else {
                                    Image(systemName: "checkmark").font(.system(size: 14, weight: .semibold))
                                    Text("Save to Profile")
                                }
                            }
                            .font(PlagitFont.bodyMedium())
                            .foregroundColor(.white)
                            .frame(maxWidth: .infinity)
                            .padding(.vertical, PlagitSpacing.lg)
                            .background(Capsule().fill(Color.plagitTeal))
                        }
                        .disabled(isSaving)
                        .padding(.top, PlagitSpacing.sm)

                        Spacer().frame(height: PlagitSpacing.xxxl)
                    }
                    .padding(.horizontal, PlagitSpacing.xl)
                }
            }
            .navigationTitle("Review Your Profile")
            .navigationBarTitleDisplayMode(.inline)
            .toolbar {
                ToolbarItem(placement: .navigationBarLeading) {
                    Button("Back") { dismiss() }
                        .font(PlagitFont.captionMedium())
                        .foregroundColor(.plagitSecondary)
                }
            }
            .onAppear { prefill() }
        }
    }

    // MARK: - Prefill from extracted data

    private func prefill() {
        let d = extractedData
        firstName = d.firstName ?? ""
        lastName = d.lastName ?? ""
        email = d.email ?? ""
        phone = d.phone ?? ""
        location = d.location ?? ""
        role = d.role ?? ""
        experience = d.experience ?? ""
        languages = d.languages ?? ""
        skills = d.skills ?? ""
        certifications = d.certifications ?? ""
        bio = d.bio ?? ""
    }

    // MARK: - Helpers

    private func fieldCard<Content: View>(_ title: String, @ViewBuilder content: () -> Content) -> some View {
        VStack(alignment: .leading, spacing: PlagitSpacing.lg) {
            Text(title).font(PlagitFont.bodyMedium()).foregroundColor(.plagitCharcoal)
            content()
        }
        .padding(PlagitSpacing.xl)
        .background(RoundedRectangle(cornerRadius: PlagitRadius.xl).fill(Color.plagitCardBackground)
            .shadow(color: PlagitShadow.subtleShadowColor, radius: PlagitShadow.subtleShadowRadius, y: PlagitShadow.subtleShadowY))
    }

    private func editField(_ placeholder: String, text: Binding<String>, icon: String, keyboard: UIKeyboardType = .default) -> some View {
        VStack(alignment: .leading, spacing: PlagitSpacing.xs) {
            HStack(spacing: PlagitSpacing.xs) {
                Image(systemName: icon).font(.system(size: 12, weight: .medium)).foregroundColor(.plagitTeal)
                Text(placeholder).font(PlagitFont.captionMedium()).foregroundColor(.plagitSecondary)
            }
            TextField(placeholder, text: text)
                .font(PlagitFont.body())
                .foregroundColor(.plagitCharcoal)
                .keyboardType(keyboard)
                .padding(.horizontal, PlagitSpacing.md)
                .padding(.vertical, PlagitSpacing.md)
                .background(RoundedRectangle(cornerRadius: PlagitRadius.md).fill(Color.plagitSurface))
        }
    }
}

File: mobile-app-flutter/lib/features/candidate/views/cv_review_view.dart
import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';
import 'package:plagit/config/app_theme.dart';

/// CV review screen — mirrors CVReviewView.swift.
class CvReviewView extends StatefulWidget {
  const CvReviewView({super.key});

  @override
  State<CvReviewView> createState() => _CvReviewViewState();
}

class _CvReviewViewState extends State<CvReviewView> {
  final _firstNameController = TextEditingController(text: 'John');
  final _lastNameController = TextEditingController(text: 'Smith');
  final _phoneController = TextEditingController(text: '+44 7700 900000');
  final _locationController = TextEditingController(text: 'London, UK');
  final _roleController = TextEditingController(text: 'Head Chef');
  final _experienceController = TextEditingController(text: '8 years');
  final _languagesController = TextEditingController(text: 'English, Italian');
  final _skillsController = TextEditingController(text: 'Fine dining, Menu planning, Team leadership');
  final _certificationsController = TextEditingController(text: 'Level 3 Food Safety, First Aid');
  final _bioController = TextEditingController(text: 'Experienced head chef with a passion for Mediterranean cuisine and team development.');

  String _jobType = 'Full-time';
  bool _isSaving = false;

  final List<String> _jobTypes = ['Full-time', 'Part-time', 'Contract', 'Freelance', 'Temporary', 'Flexible'];

  @override
  void dispose() {
    _firstNameController.dispose();
    _lastNameController.dispose();
    _phoneController.dispose();
    _locationController.dispose();
    _roleController.dispose();
    _experienceController.dispose();
    _languagesController.dispose();
    _skillsController.dispose();
    _certificationsController.dispose();
    _bioController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: AppColors.background,
      appBar: AppBar(
        backgroundColor: Colors.transparent,
        elevation: 0,
        leading: TextButton(
          onPressed: () => context.pop(),
          child: const Text('Back', style: TextStyle(fontSize: 13, fontWeight: FontWeight.w500, color: AppColors.secondary)),
        ),
        title: const Text('Review Your Profile', style: TextStyle(fontSize: 16, fontWeight: FontWeight.w600, color: AppColors.charcoal)),
        centerTitle: true,
      ),
      body: SingleChildScrollView(
        padding: const EdgeInsets.symmetric(horizontal: AppSpacing.xl),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            const SizedBox(height: AppSpacing.lg),
            // Header
            Center(
              child: Column(
                children: [
                  Container(
                    width: 56, height: 56,
                    decoration: BoxDecoration(shape: BoxShape.circle, color: AppColors.online.withValues(alpha: 0.1)),
                    child: const Icon(Icons.check_circle, size: 28, color: AppColors.online),
                  ),
                  const SizedBox(height: AppSpacing.sm),
                  const Padding(
                    padding: EdgeInsets.symmetric(horizontal: AppSpacing.xl),
                    child: Text(
                      'We extracted the details below from your CV. Please review, edit if needed, and confirm before saving.',
                      style: TextStyle(fontSize: 13, color: AppColors.secondary),
                      textAlign: TextAlign.center,
                    ),
                  ),
                  const SizedBox(height: AppSpacing.xs),
                  const Text('Nothing will be saved until you confirm.', style: TextStyle(fontSize: 10, color: AppColors.tertiary)),
                ],
              ),
            ),
            const SizedBox(height: AppSpacing.xl),
            // Personal Details
            _buildFieldCard('Personal Details', [
              _buildEditField('First Name', _firstNameController, Icons.person),
              _buildEditField('Last Name', _lastNameController, Icons.person),
              _buildEditField('Phone Number', _phoneController, Icons.phone, keyboard: TextInputType.phone),
              _buildEditField('Location', _locationController, Icons.location_on),
            ]),
            const SizedBox(height: AppSpacing.xl),
            // Professional
            _buildFieldCard('Professional', [
              _buildEditField('Job Role', _roleController, Icons.work),
              _buildEditField('Years of Experience', _experienceController, Icons.schedule),
              _buildJobTypePicker(),
            ]),
            const SizedBox(height: AppSpacing.xl),
            // Languages & Skills
            _buildFieldCard('Languages & Skills', [
              _buildEditField('Languages', _languagesController, Icons.language),
              _buildEditField('Skills', _skillsController, Icons.star),
              _buildEditField('Certifications', _certificationsController, Icons.military_tech),
            ]),
            const SizedBox(height: AppSpacing.xl),
            // Bio
            if (_bioController.text.isNotEmpty)
              _buildFieldCard('About', [
                Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Row(
                      children: const [
                        Icon(Icons.format_align_left, size: 12, color: AppColors.teal),
                        SizedBox(width: AppSpacing.xs),
                        Text('Bio', style: TextStyle(fontSize: 13, fontWeight: FontWeight.w500, color: AppColors.secondary)),
                      ],
                    ),
                    const SizedBox(height: AppSpacing.sm),
                    TextField(
                      controller: _bioController,
                      maxLines: null,
                      minLines: 3,
                      style: const TextStyle(fontSize: 15, color: AppColors.charcoal),
                      decoration: const InputDecoration(
                        border: InputBorder.none,
                        enabledBorder: InputBorder.none,
                        focusedBorder: InputBorder.none,
                        isDense: true,
                        contentPadding: EdgeInsets.zero,
                      ),
                    ),
                  ],
                ),
              ]),
            const SizedBox(height: AppSpacing.lg),
            // Note
            const Center(
              child: Text('You can edit any extracted field before saving.', style: TextStyle(fontSize: 10, color: AppColors.tertiary)),
            ),
            const SizedBox(height: AppSpacing.lg),
            // Save button
            GestureDetector(
              onTap: _isSaving ? null : _save,
              child: Container(
                width: double.infinity,
                padding: const EdgeInsets.symmetric(vertical: AppSpacing.lg),
                decoration: BoxDecoration(color: AppColors.teal, borderRadius: BorderRadius.circular(AppRadius.full)),
                alignment: Alignment.center,
                child: _isSaving
                    ? const SizedBox(width: 20, height: 20, child: CircularProgressIndicator(color: Colors.white, strokeWidth: 2))
                    : Row(
                        mainAxisSize: MainAxisSize.min,
                        children: const [
                          Icon(Icons.check, size: 14, color: Colors.white),
                          SizedBox(width: AppSpacing.sm),
                          Text('Save to Profile', style: TextStyle(fontSize: 15, fontWeight: FontWeight.w600, color: Colors.white)),
                        ],
                      ),
              ),
            ),
            const SizedBox(height: AppSpacing.xxxl),
          ],
        ),
      ),
    );
  }

  Widget _buildFieldCard(String title, List<Widget> children) {
    return Container(
      padding: const EdgeInsets.all(AppSpacing.xl),
      decoration: BoxDecoration(
        color: AppColors.cardBackground,
        borderRadius: BorderRadius.circular(AppRadius.xl),
        boxShadow: [BoxShadow(color: Colors.black.withValues(alpha: 0.03), blurRadius: 6, offset: const Offset(0, 2))],
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text(title, style: const TextStyle(fontSize: 15, fontWeight: FontWeight.w600, color: AppColors.charcoal)),
          const SizedBox(height: AppSpacing.lg),
          ...children.map((c) => Padding(padding: const EdgeInsets.only(bottom: AppSpacing.md), child: c)),
        ],
      ),
    );
  }

  Widget _buildEditField(String label, TextEditingController controller, IconData icon, {TextInputType keyboard = TextInputType.text}) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Row(
          children: [
            Icon(icon, size: 12, color: AppColors.teal),
            const SizedBox(width: AppSpacing.xs),
            Text(label, style: const TextStyle(fontSize: 13, fontWeight: FontWeight.w500, color: AppColors.secondary)),
          ],
        ),
        const SizedBox(height: AppSpacing.xs),
        Container(
          padding: const EdgeInsets.symmetric(horizontal: AppSpacing.md, vertical: AppSpacing.md),
          decoration: BoxDecoration(color: AppColors.surface, borderRadius: BorderRadius.circular(AppRadius.md)),
          child: TextField(
            controller: controller,
            keyboardType: keyboard,
            style: const TextStyle(fontSize: 15, color: AppColors.charcoal),
            decoration: InputDecoration(
              hintText: label,
              hintStyle: const TextStyle(color: AppColors.tertiary),
              border: InputBorder.none,
              enabledBorder: InputBorder.none,
              focusedBorder: InputBorder.none,
              isDense: true,
              contentPadding: EdgeInsets.zero,
            ),
          ),
        ),
      ],
    );
  }

  Widget _buildJobTypePicker() {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Row(
          children: const [
            Icon(Icons.badge, size: 12, color: AppColors.teal),
            SizedBox(width: AppSpacing.xs),
            Text('Job Type', style: TextStyle(fontSize: 13, fontWeight: FontWeight.w500, color: AppColors.secondary)),
          ],
        ),
        const SizedBox(height: AppSpacing.sm),
        SingleChildScrollView(
          scrollDirection: Axis.horizontal,
          child: Row(
            children: _jobTypes.map((type) {
              final active = _jobType == type;
              return Padding(
                padding: const EdgeInsets.only(right: AppSpacing.sm),
                child: GestureDetector(
                  onTap: () => setState(() => _jobType = type),
                  child: Container(
                    padding: const EdgeInsets.symmetric(horizontal: AppSpacing.md, vertical: AppSpacing.sm),
                    decoration: BoxDecoration(
                      color: active ? AppColors.teal : AppColors.surface,
                      borderRadius: BorderRadius.circular(AppRadius.full),
                    ),
                    child: Text(type, style: TextStyle(fontSize: 13, fontWeight: FontWeight.w500, color: active ? Colors.white : AppColors.secondary)),
                  ),
                ),
              );
            }).toList(),
          ),
        ),
      ],
    );
  }

  void _save() {
    setState(() => _isSaving = true);
    Future.delayed(const Duration(seconds: 1), () {
      if (mounted) {
        setState(() => _isSaving = false);
        context.pop();
      }
    });
  }
}
